snapcraft/internal/errors.py

```diff
@@ -73,91 +73,105 @@ def __str__(self) -> str:
         return self.get_brief()
 
 
-class MissingStateCleanError(SnapcraftError):
-    fmt = (
-        "Failed to clean: "
-        "Missing state for {step.name!r}. "
-        "To clean the project, run `snapcraft clean`."
-    )
+class SnapcraftReportableException(SnapcraftException, ABC):
+    """Helper class for reportable Snapcraft Exceptions."""
+
+    def get_reportable(self) -> bool:
+        return True
+
 
+class MissingStateCleanError(SnapcraftException):
     def __init__(self, step):
-        super().__init__(step=step)
+        self._step = step
 
+    def get_brief(self) -> str:
+        return f"Failed to clean for step {self._step.name!r}."
 
-class StepOutdatedError(SnapcraftError):
+    def get_resolution(self) -> str:
+        return "Run `snapcraft clean` and retry build."
 
-    fmt = (
-        "Failed to reuse files from previous run: "
-        "The {step.name!r} step of {part!r} is out of date:\n"
-        "{report}"
-        "To continue, clean that part's {step.name!r} step by running "
-        "`snapcraft clean {parts_names} -s {step.name}`."
-    )
 
+class StepOutdatedError(SnapcraftException):
     def __init__(
         self, *, step, part, dirty_report=None, outdated_report=None, dependents=None
```

Comment:
want to add type annotations while at it?

Reply:
Sounds like a good idea :D

Reply:
Done!